doc/decisions/simplify_api.md

```diff
@@ -0,0 +1,94 @@
+# Simplify API
+
+## Problem
+
+According to src/libs/elektra/symbols.map the public core API has 124 functions, which
+is arguable too much for a key-value API which has as highest goal simplicity.
```

Comment:
> By my count

Thank you, I added this information in 4781dcfb9437a18294de045ecd4791ade8e177d9

> As a sidenote, we should consider creating separate folders and symbols files for the libelektra-core and libelektra-kdb libraries. The current situation can be a bit confusing.

This is actually quite important: #4219 (otherwise we would get quite a mess when the rust code gets added)